### ISSUE TITLE
Unable to run websocket demo

### ISSUE THREAD

**Issue by clartaq** [luminus-framework/luminus#194]
I have tried both the `http-kit` version and the `immutant` version from the repo as well as the version of the instructions on your blog. None seem to produce a working demo.

I have added missing dependencies in the `project.clj` produced by the template. That reduced the number of errors, but still did not yield a working demo. Lots of un-resolvable symbols. For example, the routes added have undefined symbols, e.g. `websocket-routes`  is not included in the namespace declaration and I can't find `base-routes` anywhere.

I'm an experienced programmer, but a complete noob in web development.

This was attempted with Windows 10 Pro, 64-bit and Java 1.8.0_131.

**Comment by yogthos** [comment]
Hi,

I just ran the example found [here](https://github.com/luminus-framework/examples/tree/master/multi-client-ws-immutant) and it appears to be working as expected. Do you have an example project where you're having the issue.

**Comment by clartaq** [comment]
Hi,

The project you pointed to works as expected.

I don't really have an example project to show you. First, I followed the instructions on your blog [here](https://yogthos.net/posts/2015-06-11-Websockets.html). When it didn't work, I tried to look at the repository pointed to by the blog, but the link was broken.

Second, I tried to follow the instructions from the [documentation site](http://www.luminusweb.net/docs/websockets.md), but had similar problems. At that time, the link to the repository was broken as well, but I see it has been updated. Thanks for that.

I haven't gone through the updated version of the doc, but having the link to the working example will let me make it through any other issues.

This issue can be closed as far as I'm concerned.

**Comment by yogthos** [comment]
Yeah, the examples got moved and I forgot to update the links. Sorry about that. Note that some of the API in the libraries may have changed since the examples got published, I'll take a look to make sure Luminus docs are in line with the latest versions.

Thanks for bringing up the issue.

**Comment by yogthos** [comment]
Just tested with a fresh project, the instructions should be working.

**Comment by clartaq** [comment]
Hmm... Trying to construct the project from the directions still didn't work for me. I only looked through the `immutant` example [here](https://github.com/luminus-framework/luminus/blob/master/resources/md/websockets.md), but the things I found that were wrong/incomplete include:

* In the section "Adding the routes to the handler". It wasn't clear without looking at the source that you probably intended the `def` to look like:

```clojure
(def app-routes
  (routes
    websocket-routes
    (-> #'home-routes
        (wrap-routes middleware/wrap-csrf)
        (wrap-routes middleware/wrap-formats))
    (route/not-found
      (:body
        (error-page {:status 404
                     :title "page not found"})))))
```
Using the same name for the `def` that is used for the following function was confusing -- was `defn app` supposed to be replaced with this def? Was `app-routes` supposed to be ignored? *etc*.

* Also in `multi-client-ws.handler`, it should mention adding `[multi-client-ws.routes.websockets :refer [websocket-routes]]` to the namespace declaration.

* Now here's something I **really** don't understand. The example project that you sent this morning builds and runs fine as is. But when I follow the instructions, the ClojureScript client needs to have `cognitect.transit` added to the project dependencies. I used `[com.cognitect/transit-cljs "0.8.239"]`. Otherwise, `figwheel` fails. I don't understand how the project in the repository can run without adding the dependency, but it does on my system.

Ahha! The project files in the repository and that generated by the template have some substantial differences. The version in the example repository includes `cljs-ajax`, which is apparently no longer included in the template-generated project file. If I add `cljs-ajax`, with it's dependency on `transit`, and remove the explicit dependency on `transit`, the project created from the documentation will run.

* The directions should mention having to add `reagent` to the project dependencies.

Once I make the changes described above, the directions for the `immutant` version work for me.
